### PR TITLE
refactor and exception handling for dns_cli.py

### PR DESCRIPTION
**Pull request by ejayesh**
No unit tests added yet.
Refactored and added exception handling to dns_cli.py.  (Will do same to codebuild_cli.py in seperate PR)

### CODE REVIEW

File: dns_cli.py [hunk 1]
```diff
@@ -265,224 +226,286 @@ def create_hosted_zone(client, domain, start_domain, base_domain):
...
-    # breakpoint()
-    # pp.pprint(response)
 
     # add records to hosted zone to validate certificate
+    cert_arn = create_cert(acm_client, domain_client, domain, base_len)
 
     return cert_arn
 
 
+def check_aws_conn(aws_profile):
```
Comment (yusufsheiqh) [inline dns_cli.py:278]: This is nice - we definitely needed this.

File: dns_cli.py [hunk 2]
```diff
@@ -265,224 +226,286 @@ def create_hosted_zone(client, domain, start_domain, base_domain):
...
+    alias_client = session.client('iam')
+    account_name = alias_client.list_account_aliases()['AccountAliases']
+    print(f"Logged in with AWS Domain account: {account_name[0]}/{sts.get_caller_identity()['Account']}\n\
+          User: {sts.get_caller_identity()['UserId']}")
+    return session
+
+
+def check_response(response):
+    if response['ResponseMetadata']['HTTPStatusCode'] != 200:
+        print("Uknown response error from AWS")
```
Comment (yusufsheiqh) [inline dns_cli.py:313]: Minor typo here.

```suggestion
        print("Unknown response error from AWS")
```

File: dns_cli.py [hunk 3]
```diff
@@ -265,224 +226,286 @@ def create_hosted_zone(client, domain, start_domain, base_domain):
...
 
         parts.pop(0)
 
     else:
-        print(f"No hosted zone found for {domain}")
+        print(f"No hosted zone found for {domain_name}")
         return
 
 
+def run_atest():
```
Comment (yusufsheiqh) [inline dns_cli.py:505]: Better to move tests to their own directory and/or own files.

**Review by yusufsheiqh — APPROVED** [review]

File: dns_cli.py (hunk 3 above)
Comment (ejayesh) [reply dns_cli.py:505]: I forgot to take out run_atest.  Will remove

File: dns_cli.py (hunk 2 above)
Comment (ejayesh) [reply dns_cli.py:313]: good point, I will add the status code

File: dns_cli.py (hunk 3 above)
Comment (yusufsheiqh) [reply dns_cli.py]: Thank you! Resolved in https://github.com/uktrade/copilot-tools/pull/29/commits/47f03694754c5665e60347821bc51297092e9e7d.

**Review by lgarvey — APPROVED** [review]
Looks good.  

Only minor thing, we should probably use click.echo instead of print for consistency.  The nice thing is we can apply colour formatting.  Not essential, so feel free to ignore for now.

https://click.palletsprojects.com/en/7.x/utils/#ansi-colors

**Pull request merged**